src/ci/azure.rs

```diff
@@ -54,26 +61,45 @@ struct TimelineLog {
 struct TimelineRecord {
     #[serde(rename = "type")]
     type_: String,
-    log: TimelineLog,
+    id: String,
+    name: String,
+    log: Option<TimelineLog>,
     #[serde(flatten)]
     outcome: BuildOutcome,
+    #[serde(skip, default)]
+    build: u64,
+}
+
+#[derive(Debug, Deserialize)]
+struct TaskReference {
+    id: String,
+    name: String,
+    version: String,
+}
+
+impl TimelineRecord {
+    fn log(&self) -> &TimelineLog {
+        self.log.as_ref().unwrap_or_else(|| {
+            panic!("log field = None for {:?}", self);
+        })
+    }
 }
 
 impl Job for TimelineRecord {
-    fn id(&self) -> u64 {
-        unreachable!();
+    fn id(&self) -> String {
+        self.id.clone()
     }
 
     fn html_url(&self) -> String {
-        unreachable!();
+        self.log.as_ref().expect("log url").url.clone()
```

Comment:
It's possible to get an HTML URL that points directly to a job:

```
https://dev.azure.com/rust-lang/rust/_build/results?buildId=<ID>&view=logs&jobId=<JOB-UUID>
```

For example:

https://dev.azure.com/rust-lang/rust/_build/results?buildId=1791&view=logs&jobId=a6152bb1-991e-54db-b884-29fab67829cc

Reply:
I guess we link to both. I personally almost never find this URL useful, but I guess YMMV. Fixed.